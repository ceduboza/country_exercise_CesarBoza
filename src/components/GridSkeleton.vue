<template>
    <ul>
        <li v-for="(_, idx) in Array(9)" :key="idx">
            <CountryCardSkeleton />
        </li>
    </ul>
</template>

<script setup lang="ts">
import CountryCardSkeleton from 'src/modules/Countries/aplication/CountryCardSkeleton/CountryCardSkeleton.vue';
</script>