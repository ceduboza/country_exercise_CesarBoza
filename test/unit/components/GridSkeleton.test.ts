import { describe, it, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";
import GridSkeleton from "../../../src/components/GridSkeleton.vue";

describe("GridSkeleton", () => {
    let wrapper

    const findAllSkeletons = wrapper => wrapper.findAllComponents({ name: "CountryCardSkeleton" });

    const factory = () => {
        wrapper = shallowMount(GridSkeleton, {
            global: {
                stubs: {
                    CountryCardSkeleton: true,
                },
            },
        })
    }

    it("should render the grid skeleton", () => {
        factory()

        expect(wrapper.html()).toMatchSnapshot();
        expect(findAllSkeletons(wrapper).length).toBe(9);
    });
});
