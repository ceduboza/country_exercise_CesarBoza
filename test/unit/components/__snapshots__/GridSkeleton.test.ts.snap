// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GridSkeleton > should render the grid skeleton 1`] = `
"<ul>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
  <li>
    <country-card-skeleton-stub></country-card-skeleton-stub>
  </li>
</ul>"
`;
